CODE_REVIEW.md

```diff
@@ -0,0 +1,41 @@
+CODE SMELLS OR PROBLEMS :
+
+1.	In book search component, subscription is not ended for 'getAllBooks' selector after storing the search results in a variable. This could be a possible code smell or memory leak. Therefore, used async pipe in HTML as when the component gets destroyed, the async pipe unsubscribes automatically to avoid potential memory leaks.
+
+2.	Book is added and removed from the reading list even when the 'reading-list' (POST and DELETE) api fails. Updated the code that shall not allow addition or removal of book in the reading list if the API fails.
```

Comment:
This is to handle optimistic updates, so when a book is added/removed the state updates before confirmation from API. But if it fails then the failure will undo the change.